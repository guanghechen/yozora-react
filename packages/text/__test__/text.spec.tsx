import React from 'react'
import { mount, render, shallow } from 'enzyme'
import Text from '../src'


describe('test suite: Test component', () => {
  const errorLogger = jest
    .spyOn(global.console, 'error')
    .mockImplementation((...args) => {
      throw new Error(args.join(' '))
    })

  afterAll(() => {
    errorLogger.mockRestore()
  })

  it('expect render a text value', () => {
    const text = 'Hello, world!'
    const wrapper = shallow(
      <Text value={ text } />
    )
    expect(wrapper.text()).toEqual(text)
  })

  it('expect render with custom className', () => {
    const text = 'Hello, world!'
    const className = 'custom-text'
    const wrapper = shallow(
      <Text className={ className } value={ text } />
    )
    expect(wrapper.hasClass(className)).toEqual(true)
    expect(wrapper.text()).toEqual(text)
  })

  it('value is required', () => {
    for (const value of [undefined, null] as any[]) {
      expect(() => {
        mount(
          <Text value={ value } />
        )
      }).toThrow(/Failed prop type: The prop `value` is marked as required/)
    }
  })

  it('forward ref', () => {
    const ref = React.createRef<HTMLSpanElement>()
    const wrapper = mount(
      <Text ref={ ref } data-value="waw" value=""/>
    )

    const o = wrapper.getDOMNode()
    expect(o).toEqual(ref.current)
    expect(o.getAttribute('data-value')).toEqual('waw')
  })

  it('snapshot', () => {
    const wrapper = render(
      <Text
        value="Hello, world!"
        style={ { color: 'orange', fontSize: '16px' } }
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
