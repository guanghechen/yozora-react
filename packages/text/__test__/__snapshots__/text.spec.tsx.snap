// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test suite: Test component snapshot 1`] = `
.c0 {
  color: var(--md-text-color);
  font-size: var(--md-text-font-size,1em);
  font-style: var(--md-text-font-style,normal);
  font-weight: var(--md-text-font-weight,400);
}

<span
  class="c0"
  style="color:orange;font-size:16px"
>
  Hello, world!
</span>
`;
